lasio/las.py

```diff
@@ -264,16 +260,17 @@ def add_section(pattern, name, **sect_kws):
                     "~A after NULL replacement data.shape {}".format(arr.shape)
                 )
 
-                n_curves = len(self.curves)
-                n_arr_cols = len(self.curves)  # provisional pending below check
-                logger.debug("n_curves=%d ncols=%d" % (n_curves, s["ncols"]))
-                if wrap == "NO":
-                    if s["ncols"] > n_curves:
-                        n_arr_cols = s["ncols"]
-                data = np.reshape(arr, (-1, n_arr_cols))
-
-                self.set_data(data, truncate=False)
-                drop.append(s["title"])
+                if "Curves" in self.sections:
+                    n_curves = len(self.curves)
+                    n_arr_cols = len(self.curves)  # provisional pending below check
+                    logger.debug("n_curves=%d ncols=%d" % (n_curves, s["ncols"]))
+                    if wrap == "NO":
+                        if s["ncols"] > n_curves:
+                            n_arr_cols = s["ncols"]
+                    data = np.reshape(arr, (-1, n_arr_cols))
+
+                    self.set_data(data, truncate=False)
+                    drop.append(s["title"])
```

Comment:
See comment in main PR thread.

Reply:
Which comment?

Reply:
Sorry! This is my first time using Github's PR-reviewing function, so I am a bit lost ;-) I'm referring to https://github.com/kinverarity1/lasio/pull/283#pullrequestreview-270520602. I think I will try and tackle this part, as it's cleaning up my mess from the past :-p

Reply:
Ah I see, sorry I also couldn't co-relate. 
if you want to read the file with only **data** sections
One approach I could think of is to change the set_data function to check if there are no Curves and then set the array value as it is for las.data (new attribute) if there are no Curves. This could be a quicker fix, however the data returned wouldn't be always very readable.